File: problems/1380-lucky-numbers-in-a-matrix/solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> luckyNumbers(int[][] matrix) {
        List<Integer> luckyNumbers = new ArrayList<>();
        int m = matrix.length;
        int n = matrix[0].length;

        for (int i = 0; i < m; i++) {
            int minInRow = Integer.MAX_VALUE;
            int colIndex = -1;

            for (int j = 0; j < n; j++) {
                if (matrix[i][j] < minInRow) {
                    minInRow = matrix[i][j];
                    colIndex = j;
                }
            }

            boolean isMaxInCol = true;
            for (int k = 0; k < m; k++) {
                if (matrix[k][colIndex] > minInRow) {
                    isMaxInCol = false;
                    break;
                }
            }

            if (isMaxInCol) {
                luckyNumbers.add(minInRow);
            }
        }

        return luckyNumbers;
    }
}

File: problems/1380-lucky-numbers-in-a-matrix/solution.cpp
class Solution {
public:
    vector<int> luckyNumbers(vector<vector<int>>& matrix) {
        vector<int> lucky;
        int m = matrix.size();
        int n = matrix[0].size();

        for (int i = 0; i < m; ++i) {
            int min_row = matrix[i][0];
            int col_index = 0;
            for (int j = 1; j < n; ++j) {
                if (matrix[i][j] < min_row) {
                    min_row = matrix[i][j];
                    col_index = j;
                }
            }

            bool is_lucky = true;
            for (int k = 0; k < m; ++k) {
                if (matrix[k][col_index] > min_row) {
                    is_lucky = false;
                    break;
                }
            }

            if (is_lucky) {
                lucky.push_back(min_row);
            }
        }

        return lucky;
    }
};